File: smart-waste-ui/src/types/truck.ts
export type TruckStatus = "available" | "in_service" | "maintenance";

export interface TruckDriver {
  id: number;
  username: string;
  full_name: string;
  email: string;
  role: string;
  is_active: boolean;
  created_at: string;
}

export interface Truck {
  id: number;
  license_plate: string;
  model: string;
  capacity_bins: number | null;
  status: TruckStatus;
  assigned_driver_id: number | null;
  assigned_driver: TruckDriver | null;
}

export interface TruckCreate {
  license_plate: string;
  model: string;
  capacity_bins?: number | null;
  status?: TruckStatus;
  assigned_driver_id?: number | null;
}

export interface TruckUpdate {
  license_plate?: string;
  model?: string;
  capacity_bins?: number | null;
  status?: TruckStatus;
  assigned_driver_id?: number | null;
}

File: backend/app/schemas/truck.py
from pydantic import BaseModel, Field
from typing import Optional
from app.models.truck import TruckStatus
from app.schemas.user import UserResponse


class TruckBase(BaseModel):
    license_plate: str = Field(..., max_length=20)
    model: str = Field(..., max_length=100)
    capacity_bins: Optional[int] = Field(None, ge=1)
    status: TruckStatus = TruckStatus.AVAILABLE


class TruckCreate(TruckBase):
    assigned_driver_id: Optional[int] = None


class TruckUpdate(BaseModel):
    license_plate: Optional[str] = Field(None, max_length=20)
    model: Optional[str] = Field(None, max_length=100)
    capacity_bins: Optional[int] = Field(None, ge=1)
    status: Optional[TruckStatus] = None
    assigned_driver_id: Optional[int] = None


class TruckResponse(TruckBase):
    id: int
    assigned_driver_id: Optional[int] = None
    assigned_driver: Optional[UserResponse] = None

    class Config:
        from_attributes = True


class TruckAssignRequest(BaseModel):
    driver_id: Optional[int] = None
